### ISSUE TITLE
Black screen after exit fullscreen

### ISSUE THREAD

**Issue by chladto1** [androidx/media#1293]
### Version

Media3 1.2.1

### Devices that reproduce the issue

Probably any device or emulator with Android 13.

### Devices that do not reproduce the issue

Any device or emulator except Android 13.

### Reproducible in the demo app?

No

### Reproduction steps

We've built our SDK around Exo/Media3 and we support two mechanisms for fullscreen.

The first method doesn't necessitate any code alterations in the client's application. When a request to enter fullscreen mode occurs, the `ViewGroup` containing the player's `SurfaceView` and other elements is removed from a `Fragment` and added to a new fullscreen `Activity`. When a user wants to exit fullscreen mode, the `Activity` is finished and the `ViewGroup` is returned to its original `Fragment`. This approach functioned correctly in Media3 version 1.1.1, but it's dysfunctional in version 1.2.1. Upon exiting fullscreen mode, only a black screen appears, accompanied by some GL errors in Logcat.

```
2024-04-17 23:21:13.532 EGL_emulation            D  eglCreateContext: 0x78042b23aad0: maj 3 min 0 rcv 3
2024-04-17 23:21:13.596 EGL_emulation            D  eglMakeCurrent: 0x78042b23aad0: ver 3 0 (tinfo 0x78063e52d300) (first time)
2024-04-17 23:21:13.606 emuglGLESv2_enc          E  device/generic/goldfish-opengl/system/GLESv2_enc/GL2Encoder.cpp:s_glDeleteProgram:2003 GL error 0x501 condition [!isShaderOrProgramObject]
2024-04-17 23:21:13.606 emuglGLESv2_enc          E  device/generic/goldfish-opengl/system/GLESv2_enc/GL2Encoder.cpp:s_glDeleteShader:2117 GL error 0x501 condition [!isShaderOrProgramObject && !isShader]
2024-04-17 23:21:13.606 emuglGLESv2_enc          E  device/generic/goldfish-opengl/system/GLESv2_enc/GL2Encoder.cpp:s_glDeleteShader:2117 GL error 0x501 condition [!isShaderOrProgramObject && !isShader]
2024-04-17 23:21:13.617 EGL_emulation            E  eglQueryContext 32c0  EGL_BAD_ATTRIBUTE
2024-04-17 23:21:13.618 EGL_emulation            E  tid 6979: eglQueryContext(2122): error 0x3004 (EGL_BAD_ATTRIBUTE)
2024-04-17 23:21:13.618 emuglGLESv2_enc          E  device/generic/goldfish-opengl/system/GLESv2_enc/GL2Encoder.cpp:s_glDeleteProgram:2003 GL error 0x501 condition [!isShaderOrProgramObject]
2024-04-17 23:21:13.618 emuglGLESv2_enc          E  device/generic/goldfish-opengl/system/GLESv2_enc/GL2Encoder.cpp:s_glDeleteShader:2117 GL error 0x501 condition [!isShaderOrProgramObject && !isShader]
2024-04-17 23:21:13.618 emuglGLESv2_enc          E  device/generic/goldfish-opengl/system/GLESv2_enc/GL2Encoder.cpp:s_glDeleteShader:2117 GL error 0x501 condition [!isShaderOrProgramObject && !isShader]
2024-04-17 23:21:13.644 SurfaceSyncer            E  Failed to find sync for id=0
2024-04-17 23:21:13.660 SurfaceUtils             D  connecting to surface 0x78051b23ef90, reason connectToSurface
2024-04-17 23:21:13.660 MediaCodec               I  [c2.goldfish.h264.decoder] setting surface generation to 1856519
2024-04-17 23:21:13.660 SurfaceUtils             D  disconnecting from surface 0x78051b23ef90, reason connectToSurface(reconnect)
2024-04-17 23:21:13.660 SurfaceUtils             D  connecting to surface 0x78051b23ef90, reason connectToSurface(reconnect)
2024-04-17 23:21:13.662 Codec2-Out...ufferQueue  D  remote graphic buffer migration 13/13
2024-04-17 23:21:13.674 Codec2Client             D  setOutputSurface -- failed to set consumer usage (6/BAD_INDEX)
2024-04-17 23:21:13.674 Codec2Client             D  setOutputSurface -- generation=1856519 consumer usage=0x900
2024-04-17 23:21:13.681 Codec2Client             D  Surface configure completed
2024-04-17 23:21:13.684 SurfaceUtils             D  disconnecting from surface 0x78051b261090, reason disconnectFromSurface
2024-04-17 23:21:13.701 ttplayer.sample          W  [SurfaceTexture-1-1813-7] bindTextureImage: clearing GL error: 0x501
```

### Expected result

Video content is visible

### Actual result

Black screen with audio only

### Media

Any content

### Bug Report

- [ ] You will email the zip file produced by `adb bugreport` to android-media-github@google.com after filing this issue.

**Comment by tonihei** [comment]
Do you have a reproduction project or steps we can follow to see this issue ourselves? 

Could you also clarify whether this is using Media3's `PlayerView` or custom view? And whether you handle the surface lifecycle yourself or let ExoPlayer handle it (by using `ExoPlayer.setVideoSurfaceView`)?

There was a [surface related change in 1.2.0 in `PlayerView`](https://github.com/androidx/media/commit/69ffac28bb77bc7faeb0b8d84ed4fe171103265e), but that should only affect devices running on API 34+ (= Android 14 and above)

**Comment by chladto1** [comment]
We are using a custom `View` and our structure is `Fragment` > `FrameLayout` > `AspectRatioFrameLayout` > `SurfaceView`. `ExoPlayer.setVideoSurfaceView` is called in embedded mode only. When transitioning into fullscreen mode, root `FrameLayout` is removed from the `Fragment` and inserted into new `Fragment`.

**Comment by chladto1** [comment]
I am calling `ExoPlayer.setVideoSurfaceView`. Just to remind, everything is working fine in Exo from v2.10.6 to v2.19.1 and also in Media3 v1.1.1, but it is broken in Media3 v1.2.1.

**Comment by tonihei** [comment]
> everything is working fine in Exo from v2.10.6 to v2.19.1 and also in Media3 v1.1.1, but it is broken in Media3 v1.2.1.

Thanks for the reminder, I missed this part. Could you let us know how to reproduce the issue then and we can take a look?

**Comment by chladto1** [comment]
I found the issue. When we updated Media3, we also had to update `targetSdk`. The issue arises from `Activity.overridePendingTransition`. The method with `enterAnim` and `exitAnim` parameters is deprecated in API level 33, and the replacement is a variant with three parameters, where the last one is `backgroundColor`. It seems that this method causes the issue only on Android 13, but it was fixed in Android 14. The fix is simply to use the deprecated variant of the method on Android 13.
